client/src/actions/embedding.ts

```diff
@@ -55,8 +55,15 @@ async function _switchEmbedding(
 export const layoutChoiceAction: ActionCreator<
   ThunkAction<Promise<void>, RootState, never, Action<"set layout choice">>
 > =
-  (newLayoutChoice: string) =>
+  (newLayoutChoice: string, isSidePanel = false) =>
   async (dispatch: AppDispatch, getState: GetState): Promise<void> => {
+    if (isSidePanel) {
+      dispatch({
+        type: "set panel embedding layout choice",
+        layoutChoice: newLayoutChoice,
+      });
+      return;
+    }
```

Comment:
Skip crossfilter and annomatrix changes on sidepanel layout choice. We just need to change the layout